web/js/components/dateline/line.js

```diff
@@ -15,136 +16,126 @@ const lineStyles = {
   },
 };
 
-class Line extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      overlay: '',
-      hovered: false,
-      textActive: props.alwaysShow,
-    };
-    this.nodeRef = React.createRef();
-  }
+export default function Line (props) {
+  const {
+    id,
+    alwaysShow,
+    date,
+    lineX,
+    lineY,
+    isCompareActive,
+    style,
+    map,
+    height,
+    setTextCoords,
+    textCoords,
+    hideText,
+  } = props;
 
-  componentDidMount() {
-    const { lineX, map } = this.props;
-    const overlay = new OlOverlay({
-      element: this.nodeRef.current,
+  const [overlay, setOverlay] = useState('');
+  const [hovered, toggleHovered] = useState(false);
+  const [textActive, toggleTextActive] = useState(alwaysShow);
+  const prevAlwaysShow = usePrevious(alwaysShow);
+
+  const nodeRef = useRef();
+  const timerRef = useRef();
+
+  useEffect(() => {
+    const newOverlay = new OlOverlay({
+      element: nodeRef.current,
       stopEvent: false,
     });
-    overlay.setPosition([lineX, 90]);
-    map.addOverlay(overlay);
-    this.setState({ overlay });
-  }
+    newOverlay.setPosition([lineX, 90]);
+    map.addOverlay(newOverlay);
+    setOverlay(newOverlay);
+  }, []);
 
-  componentDidUpdate(prevProps) {
-    const { lineX, lineY, alwaysShow } = this.props;
-    const { overlay } = this.state;
-    overlay.setPosition([lineX, lineY]);
-    if (!alwaysShow && alwaysShow !== prevProps.alwaysShow) {
-      // eslint-disable-next-line react/no-did-update-set-state
-      this.setState({ textActive: false });
+  useEffect(() => {
+    if (overlay !== '') return overlay.setPosition([lineX, lineY]);
+    if (!alwaysShow && alwaysShow !== prevAlwaysShow) {
+      toggleTextActive(true);
     }
-  }
-
-  mouseOver = () => {
-    this.setState({
-      hovered: true,
-    });
-  }
+  });
 
-  mouseOut = () => {
-    this.setState({
-      hovered: false,
-    });
-  }
+  const mouseOver = () => {
+    toggleHovered(true);
+  };
 
-  mouseOverHidden = (e) => {
-    const { map, setTextCoords } = this.props;
-    setTextCoords(map.getCoordinateFromPixel([e.clientX, e.clientY]));
-    this.setState({
-      textActive: true,
-    });
-  }
+  const mouseOut = () => {
+    toggleHovered(false);
+  };
 
-  mouseOutHidden = () => {
-    this.setState({ textActive: false });
-  }
+  const mouseOverHidden = (e) => {
+    const coords = map.getCoordinateFromPixel([e.clientX, e.clientY]);
+    timerRef.current = setTimeout(() => {
+      setTextCoords(coords);
+      toggleTextActive(true);
+    }, 500);
```

Comment:
After playing around with this some I think we can drop this timeout to 300ms.  Otherwise this looks good to go 👍🏻